src/app/home-page/new-dashboard/new-dashboard.component.ts

```diff
@@ -0,0 +1,48 @@
+import { Component, OnInit } from '@angular/core';
+import { Base } from 'app/shared/base';
+import { Observable } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+import { User } from '../../shared/swagger/model/user';
+import { UserQuery } from '../../shared/user/user.query';
+import { RegisterToolService } from 'app/container/register-tool/register-tool.service';
+import { MatDialog } from '@angular/material/dialog';
+import { RegisterToolComponent } from 'app/container/register-tool/register-tool.component';
+import { AlertService } from 'app/shared/alert/state/alert.service';
+
+@Component({
+  selector: 'app-new-dashboard',
+  templateUrl: './new-dashboard.component.html',
+  styleUrls: ['./new-dashboard.component.scss'],
+})
+export class NewDashboardComponent extends Base implements OnInit {
+  public user$: Observable<User>;
+  user: User;
+
+  constructor(
+    private userQuery: UserQuery,
+    private registerToolService: RegisterToolService,
+    private dialog: MatDialog,
+    private alertService: AlertService
+  ) {
+    super();
+  }
+
+  ngOnInit() {
+    this.userQuery.user$.pipe(takeUntil(this.ngUnsubscribe)).subscribe((user) => {
+      this.user = user;
+    });
+    this.registerToolService.isModalShown.pipe(takeUntil(this.ngUnsubscribe)).subscribe((isModalShown: boolean) => {
```

Comment:
I'd never noticed this before, but it's odd... entry-box.component.ts sets isModalShown to true, then this component opens the dialog.

Without revisiting the whole thing, which I'm not quite grokking (I know it was like this), it seems like this code should be in entry-box.component.ts, because that's the one thats isModalShown to true.

Reply:
I initially put it in the entry-box.component.ts file but I ran into some issues that resulted in me needing to put it here (to be honest i don't remember what the exact errors were but I can try it again to check it out)